code/citadel/crew_objectives/cit_crewobjectives_security.dm

```diff
@@ -0,0 +1,119 @@
+
+/datum/objective/crew/headofsecurity/datfukkendisk/check_completion()
+	if(owner.current && owner.current.check_contents_for(/obj/item/disk/nuclear) && SSshuttle.emergency.shuttle_areas[get_area(owner.current)])
+		return 1
+	else
+		return 0
+
+/datum/objective/crew/headofsecurity/enjoyyourstay
+	explanation_text = "Welcome to Space Station 13. Enjoy your stay."
+
+/datum/objective/crew/headofsecurity/enjoyyourstay/New()
+	. = ..()
+	update_explanation_text()
+
+/datum/objective/crew/headofsecurity/enjoyyourstay/update_explanation_text()
+	. = ..()
+	var/list/edglines = list("Welcome to Space Station 13. Enjoy your stay.", "You signed up for this.", "Abandon hope.", "The tide's gonna stop eventually.", "Hey, someone's gotta do it.", "No, you can't resign.")
+	explanation_text = pick(edglines)
+
+/datum/objective/crew/headofsecurity/enjoyyourstay/check_completion()
+	explanation_text = "Enforce Space Law at all costs."
+	if(owner.current)
+		var/mob/living/carbon/human/H = owner.current
+		var/obj/item/card/id/theID = H.get_idcard()
+		if(istype(theID))
+			if(!H.get_assignment() == "Head of Security")
+				return 0
+	return 1
+
+/datum/objective/crew/securityofficer/
+
+/datum/objective/crew/securityofficer/enjoyyourstay
+	explanation_text = "Welcome to Space Station 13. Enjoy your stay."
+
+/datum/objective/crew/securityofficer/enjoyyourstay/New()
+	. = ..()
+	update_explanation_text()
+
+/datum/objective/crew/securityofficer/enjoyyourstay/update_explanation_text()
+	. = ..()
+	var/list/edglines = list("Welcome to Space Station 13. Enjoy your stay.", "You signed up for this.", "Abandon hope.", "The tide's gonna stop eventually.", "Hey, someone's gotta do it.", "No, you can't resign.")
+	explanation_text = pick(edglines)
+
+/datum/objective/crew/securityofficer/enjoyyourstay/check_completion()
+	explanation_text = "Enforce Space Law at all costs."
+	if(owner.current)
+		var/mob/living/carbon/human/H = owner.current
+		var/obj/item/card/id/theID = H.get_idcard()
+		if(istype(theID))
+			if(!H.get_assignment() == "Security Officer")
+				return 0
+	return 1
+
+/datum/objective/crew/warden/
+
+/datum/objective/crew/warden/enjoyyourstay
+	explanation_text = "Welcome to Space Station 13. Enjoy your stay."
+
+/datum/objective/crew/warden/enjoyyourstay/New()
+	. = ..()
+	update_explanation_text()
+
+/datum/objective/crew/warden/enjoyyourstay/update_explanation_text()
+	. = ..()
+	var/list/edglines = list("Welcome to Space Station 13. Enjoy your stay.", "You signed up for this.", "Abandon hope.", "The tide's gonna stop eventually.", "Hey, someone's gotta do it.", "No, you can't resign.")
+	explanation_text = pick(edglines)
+
+/datum/objective/crew/warden/enjoyyourstay/check_completion()
+	explanation_text = "Enforce Space Law at all costs."
+	if(owner.current)
+		var/mob/living/carbon/human/H = owner.current
+		var/obj/item/card/id/theID = H.get_idcard()
+		if(istype(theID))
+			if(!H.get_assignment() == "Warden")
```

Comment:
same as before have the entire thing in a ! statement instead of just get assignment to prevent issues I'm pretty sure I've ran into issues in the past by doing this .